### PR TITLE
Support for robotics tasks in control_env + MPPI Bug fix

### PR DESCRIPTION
**Pull request by freiberg-roman**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Motivation and Context / Related issue

There was a small bug in the MPPI implementation concerning the noise. For future planners, it might be also of interest to have harder environments, so I added a few lines to support the robotics tasks

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](../../CONTRIBUTING.md) document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

**Comment by luisenp** [comment]
Hi @freiberg-roman. Any comments on the above question?

**Comment by freiberg-roman** [comment]
Hello @luisenp. Sorry, I am short on time at the moment. The reward in HandManipulateBlock and similar robotics control tasks is the negative position and rotation distance, so the optimal reward would be 0. Currently, there is a small problem with the reward calculation, as the done signal results in a reward of 0 for the rest of the trajectory, which motivates planners to drop the cube. The simplest solution, also proposed in PDDM, is to add a large terminal penalty. I will push an update for this soon :)

**Comment by luisenp** [comment]
HI @freiberg-roman, should I wait for your fix before merging this branch? Is there any other test I can do in the mean time?

File: mbrl/util/mujoco.py [hunk 1]
```diff
@@ -217,9 +219,20 @@ def __init__(self, env: gym.wrappers.TimeLimit):
         elif "mbrl.third_party.dmc2gym" in self._env.env.__class__.__module__:
             self._enter_method = self._enter_dmcontrol
             self._exit_method = self._exit_dmcontrol
+        elif "gym.envs.robotics" in self._env.env.__class__.__module__:
```
Comment (luisenp) [inline mbrl/util/mujoco.py:222]: I'm observing some weird behavior where the evaluation function starts returning trajectories with all 0 rewards, yet this is not reflected in the actual evaluated trajectory. Have you seen this? Seems to be related to having multiple processes, maybe the environments are sharing some resource?

File: mbrl/util/mujoco.py [hunk 2]
```diff
@@ -156,6 +156,8 @@ def make_env_from_str(env_name: str) -> gym.Env:
         env = dmc2gym.make(domain_name=domain, task_name=task)
     elif "gym___" in env_name:
         env = gym.make(env_name.split("___")[1])
+    elif "robo___" in env_name:
+        env = gym.make(env_name.split("___")[1], reward_type="dense")
```
[the remark this answers is not in the document]
Comment (freiberg-roman) [reply mbrl/util/mujoco.py:160]: good idea!

File: mbrl/util/mujoco.py (hunk 1 above)
Comment (freiberg-roman) [reply mbrl/util/mujoco.py:223]: Thanks! That would explain a lot. I'll have to look into that.

**Review by luisenp — COMMENTED** [review]
Left one more minor comment. Also , we could consider adding the robo option to method `make_env` in line 90 as well. 

Finally, [this](https://github.com/martius-lab/iCEM/blob/main/icem/environments/robotics.py#L31) might be useful to debug the env rollout for the robotics environments.

File: mbrl/util/mujoco.py [hunk 3]
```diff
@@ -217,9 +220,20 @@ def __init__(self, env: gym.wrappers.TimeLimit):
...
             self._enter_method = self._enter_dmcontrol
             self._exit_method = self._exit_dmcontrol
+        elif "gym.envs.robotics" in self._env.env.__class__.__module__:
+            self._enter_method = self._enter_mujoco_gym_robo
+            self._exit_method = self._exit_mujoco_gym_robo
         else:
             raise RuntimeError("Tried to freeze an unsupported environment.")
 
+    def _enter_mujoco_gym_robo(self):
+        self._init_state = self._env.unwrapped.sim.get_state()
```
Comment (luisenp) [inline mbrl/util/mujoco.py:230]: This is a guess, but try with making `self._init_state` a copy of the output of `get_state()`.

Comment (freiberg-roman) [reply mbrl/util/mujoco.py]: I have tried something similar. Can you check if this is working properly now?

Comment (freiberg-roman) [reply mbrl/util/mujoco.py]: I have found that the problem unfortunately still remains..

Comment (luisenp) [reply mbrl/util/mujoco.py]: That's unfortunate. I'll take another look soon, thanks for checking.

**Pull request closed without merging**